kamaki/clients/compute/__init__.py

```diff
@@ -441,3 +441,56 @@ def delete_floating_ip(self, tenant_id, fip_id=None):
         """
         r = self.floating_ips_delete(tenant_id, fip_id)
         return r.headers
+
+    def list_keypairs(self):
+        """
+        :returns: (list) [
+            {id: ..., name: ..., public_key: ..., fingerprint: ...},
+            ... ]
+        """
+        r = self.keypairs_get()
+        return r.json['keypairs']
+
+    def get_keypair_details(self, key_name=None):
+        """
+        :param key_name: (str) The name of the key
+
+        :returns: (dict) {keypair: ..., id: ..., name: ..., public_key: ...,
+                                   fingerprint: ...., created_at: ...,
+                                   updated_at: ...}
+        """
+        r = self.keypairs_get(key_name=key_name)
+        return r.json['keypair']
+
+    def create_or_update_key(self, key_name=None, public_key=None):
```

Comment:
First of all, the string generator method is not implemented (yet?), therefore `key_name` should be obligatory (until the feature is implemented). Each commit should be as self-consistent as possible. Therefore, we will make the argument optional only if and when we add the name generator in the method.

Second, I agree with @skalkoto, we should keep the client library simple. Adding a random string generator increases the complexity of the semantics. In other words, the answer to the question "what does this method do" is simpler if it does NOT contain a name generator.